src/createServer.js

```diff
@@ -1,8 +1,97 @@
 'use strict';
 
+const http = require('http');
+const fs = require('fs');
+const zlib = require('zlib');
+const formidable = require('formidable');
+const { pipeline } = require('stream');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = new http.Server();
+
+  server.on('request', (req, res) => {
+    if (req.url === '/' && req.method === 'GET') {
+      res.statusCode = 200;
+      res.setHeader('Content-Type', 'text/html');
+      fs.createReadStream('./public/index.html').pipe(res);
+
+      return;
+    }
+
+    if (req.url === '/compress' && req.method === 'GET') {
```

Comment:
The message from the error and the test are not aligned here.
Your are testing if the method is `GET`, and the error says you allow **only `POST`**.
So, what's the problem? If someone sent a `PUT`, would this test block it? Nope.
Is `PUT` the same as `POST`? Also nope.

So, the test you are looking for is: `req.method !== 'POST'`